homeassistant/components/automation/config.py

```diff
@@ -110,15 +120,15 @@ async def _try_async_validate_config_item(hass, config, full_config=None):
         async_log_exception(ex, DOMAIN, full_config or config, hass)
         return None
 
-    if isinstance(config, blueprint.BlueprintInputs):
-        return config
+    if isinstance(valid_config, blueprint.BlueprintInputs):
+        return valid_config
 
-    config = AutomationConfig(config)
-    config.raw_config = raw_config
-    return config
+    automation_config = AutomationConfig(config)
```

Comment:
`AutomationConfig` needs the `validated_config` not the original one. That seems to be the reason why so many tests fail.